clang/test/SemaSYCL/warn-constexpr-kernel-arg.cpp

```diff
@@ -0,0 +1,38 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -internal-isystem %S/Inputs -fsyntax-only -Wsycl-strict -sycl-std=2020 -verify %s
+
+#include "sycl.hpp"
+
+using namespace cl::sycl;
+
+queue q;
+
+class LambdaKernel;
+
+int main() {
+
+  constexpr unsigned V = 16;
+  struct Id {
+    int x;
+
+    Id(int x) : x(x) {}
+
+    Id operator*(const unsigned &val) const {
+      return Id(x * val);
+    }
+
+    operator int() const {
+      return x;
+    }
+  };
+
+  Id i{5};
+
+  q.submit([&](handler &h) {
+    // expected-note@+1 {{in instantiation of function template specialization}}
+    h.single_task<LambdaKernel>(
+        [=]() {
+          int A = i * V; // expected-warning {{constexpr 'V' is captured and is not a compile-time constant}}
```

Comment:
Do you need 'Id' for this test? Can the test be made simpler?

Reply:
The point is to make constexpr odr-used. In this test it is odr-used because operator* takes a reference. Any other test which makes constexpt odr-used will also work here.

Reply:
Was it resolved?

Reply:
This was marked resolved cause the example makes ODR-use of `constexpr` var 'V' which makes lambda to capture it. So the struct Id wasn't added without a reason. But I can try to make the test case simpler and update.

Reply:
> But I can try to make the test case simpler and update.

If I understand correctly, @elizabethandrews were asking for it.

Reply:
I didn't understand the point of this test initially. After @Fznamznon's explanation, I am okay with this if none of the other reviewers have an issue . However, wouldn't a simple pointer assignment in kernel also achieve required test?
`const unsigned *ptr = &V;`

Reply:
`const unsigned *ptr = &V;` is ODR-used as well. Updated a simplified version of the test.